### ISSUE TITLE
Magit (transient) not reading values file when location is custom

### ISSUE THREAD

**Issue by mtwomey** [hlissner/doom-emacs#5925]
### What did you expect to happen?

I update my transient variables:

```
  (setq transient-values-file "~/.doom.d/.transient/values"
       transient-history-file "~/.doom.d/.transient/history"
       transient-levels-file "~/.doom.d/.transient/levels"))
```

Now the saved transient preferences in magit are saved and loaded from these files.

### What actually happened?

This works for _writing_. Saving preferences in magit (with C-X C-S) saves values to this location (in the values file). However after restarting emacs the values are not read in magit. Note: the variables are still successfully set to this custom location, the file is simply not being used.

### Describe your attempts to resolve the issue

Tried setting these values at different times, was unable to find a setup that worked.

### Steps to reproduce

1. Set the values to a different custom location
1. Save some new preferences
1. Quite and restart emacs
1. Observe if the preferences are taking effect in magit

### System Information

Latest pull from github

**Comment by mtwomey** [comment]
The problem here is that I cannot stop this from happening in the doom magit module:

```
(use-package! magit
  :commands magit-file-delete
  :defer-incrementally (dash f s with-editor git-commit package eieio transient)
  :init
  (setq magit-auto-revert-mode nil)  ; we do this ourselves further down
  ;; Must be set early to prevent ~/.emacs.d/transient from being created
  (setq transient-levels-file  (concat doom-etc-dir "transient/levels")
        transient-values-file  (concat doom-etc-dir "transient/values")
        transient-history-file (concat doom-etc-dir "transient/history"))
```
It sets the values then trigger transient. So if I set them before the magit module, they get overridden. If I set them after, it's too late as transient has already loaded.

How can I work around this?

**Comment by Lythenas** [comment]
Using `after!` will probably work. E.g.
```elisp
(after! magit
  (setq ...))
```

The `:init` section in `use-package!` gets run before the package is loaded and the code in `after!` gets run after the package is loaded so it should overwrite the values correctly. (But I don't known when exactly transient is triggered.)

**Comment by mtwomey** [comment]
@tarsius So I'm trying this here in my own init.el file:

```
(after! magit
  (setq transient-values-file "~/.doom.d/.transient/values.el"
        transient-history-file "~/.doom.d/.transient/history.el"
        transient-levels-file "~/.doom.d/.transient/levels.el"))
```

When I initially load emacs and immediately inspect the `transient-values-file` variable, it is correct (` ~/.doom.d/.transient/values.el`). However in Magit, it's still not using those value.

In comparison, if I directly edit the values in `.emacs.d/modules/tools/magit/config.el` everything works fine.

Obviously this is a bad solution as I'd have to continually remember to maintain it every time I update.

Note - the doom emacs magit package sets these values in it's `:init` section. So doing something in my own config using `after!` is too late.

**Comment by mtwomey** [comment]
Yes agree and also make them pull from variables  set outside of the module which could still have those default values (but could be otherwise customized).

**Comment by mtwomey** [comment]
Still hoping for some workaround here. Right now I'm still patching the core Doom install after every update...
